src/components/Square.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+const Square = ({
+  i,
+  squares,
+  setSquares,
+  remainingAlphabet,
+  setRemainingAlphabet,
+  errorMessage,
+  setErrorMessage,
+  numberSelected,
+}) => {
+  const editInput = e => {
+    const newSquares = JSON.parse(JSON.stringify(squares));
+    const workRemainingAlphabet = JSON.parse(JSON.stringify(remainingAlphabet));
+    let newLetter = e.target.value.replace(/[^a-z]/gi, '').toUpperCase();
+    // Check to see if you have reached the extent of your letter useage
+    let workErrorMessage = '';
+    if (
+      numberSelected < 27 - remainingAlphabet.length + 1 &&
+      ['A', 'E', 'I', 'O', 'U'].indexOf(newLetter) === -1
+    ) {
+      workRemainingAlphabet.push('');
+      workErrorMessage = 'You have reached the extent of your letter useage... please start over';
+      newLetter = '';
+    }
+    setErrorMessage(workErrorMessage);
+    // Ensure input is a letter and if it is save it
+    // Ensure letter is available. If not generate ab error message
+    if (
+      workRemainingAlphabet.indexOf(e.target.value.toUpperCase()) === -1 &&
+      e.target.value !== '' &&
+      workErrorMessage === '' &&
+      ['A', 'E', 'I', 'O', 'U'].indexOf(newLetter) === -1
```

Comment:
This check seems to be occurring a lot, could you make it into a helper?

Reply:
Wasn't quite sure what you wanted here.... so I replaced the array with a variable name. The indexOf doesn't always get the same parm and this check is always paired with other checks.

Reply:
Created a utility to replace the entire phrase.